tests/cases/conformance/classes/members/privateNames/typeFromPrivatePropertyAssignment.ts

```diff
@@ -0,0 +1,13 @@
+// @target: esnext
+
+type Foo = { foo?: string };
+
+class C {
+    #a?: Foo;
+    #b?: Foo;
+
+    m() {
+        const a = this.#a || {};
+        this.#b = this.#b || {};
+    }
+}
```

Comment:
Let's add a `.js` file to this test, since that code path ideally wouldn't have occurred in a TS file in the first place.

```suggestion
// @filename: a.ts
type Foo = { foo?: string };

class C {
    #a?: Foo;
    #b?: Foo;

    m() {
        const a = this.#a || {};
        this.#b = this.#b || {};
    }
}

// @filename: b.js
class C {
    /** @type {{ foo?: string } | undefined } */
    #a;
    /** @type {{ foo?: string }| undefined } */
    #b;

    m() {
        const a = this.#a || {};
        this.#b = this.#b || {};
    }
}
```